addons/product/models/product_template.py

```diff
@@ -410,19 +414,8 @@ def create(self, vals_list):
         # This is needed to set given values to first variant after creation
         for template, vals in zip(templates, vals_list):
             related_vals = {}
-            if vals.get('barcode'):
-                related_vals['barcode'] = vals['barcode']
-            if vals.get('default_code'):
-                related_vals['default_code'] = vals['default_code']
-            if vals.get('standard_price'):
-                related_vals['standard_price'] = vals['standard_price']
-            if vals.get('volume'):
-                related_vals['volume'] = vals['volume']
-            if vals.get('weight'):
-                related_vals['weight'] = vals['weight']
-            # Please do forward port
-            if vals.get('packaging_ids'):
-                related_vals['packaging_ids'] = vals['packaging_ids']
+            for field_name in [x for x in self._get_related_fields_variant_template() if x in vals.keys()]:
```

Comment:
```suggestion
            for field_name in set(self._get_related_fields_variant_template()) & vals.keys():
```

And even better, `_get_related_fields_variant_template` could directly return a set :)

Reply:
thanks for your review. 
regarding this one, ``self._get_related_fields_variant_template() & vals.keys()`` seems to work in python3. (even if ``_get_related_fields_variant_template`` return a list).

see : ``list(['a']) & {'a': 2, 'b': 3}.keys()``